drivers/gpio/gpio_nct38xx.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_nct38xx_gpio
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/gpio/gpio_nct38xx.h>
+#include <kernel.h>
+#include <sys/util_macro.h>
+
+#include "gpio_nct38xx.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(gpio_ntc38xx, CONFIG_GPIO_LOG_LEVEL);
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) ((const struct gpio_nct38xx_config *)(dev)->config)
+#define DRV_DATA(dev) ((struct gpio_nct38xx_data *)(dev)->data)
+
+void nct38xx_gpio_alert_handler(const struct device *dev)
+{
+	const struct gpio_nct38xx_config *const config = DRV_CONFIG(dev);
+
+	for (int i = 0; i < config->sub_gpio_port_num; i++) {
+		gpio_nct38xx_dispatch_port_isr(config->sub_gpio_dev[i]);
+	}
+}
+
+static int nct38xx_init_interrupt(const struct device *dev)
+{
+	uint16_t alert, alert_mask = 0;
+
+	/* Disable all interrupt */
+	if (nct38xx_reg_burst_write(dev, NCT38XX_REG_ALERT_MASK, (uint8_t *)&alert_mask,
+				    sizeof(alert_mask))) {
+		return -EIO;
+	}
+
+	/* Enable vendor-defined alert for GPIO. */
+	alert_mask |= BIT(NCT38XX_REG_ALERT_MASK_VENDOR_DEFINDED_ALERT);
+
+	/* Clear alert */
+	if (nct38xx_reg_burst_read(dev, NCT38XX_REG_ALERT, (uint8_t *)&alert, sizeof(alert))) {
+		return -EIO;
+	}
+	alert &= alert_mask;
+	if (alert) {
+		if (nct38xx_reg_burst_write(dev, NCT38XX_REG_ALERT, (uint8_t *)&alert,
+					    sizeof(alert))) {
+			return -EIO;
+		}
+	}
+
+	if (nct38xx_reg_burst_write(dev, NCT38XX_REG_ALERT_MASK, (uint8_t *)&alert_mask,
+				    sizeof(alert_mask))) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int nct38xx_gpio_init(const struct device *dev)
+{
+	const struct gpio_nct38xx_config *const config = DRV_CONFIG(dev);
+
+	/* Check I2C is ready  */
+	if (!device_is_ready(config->i2c_dev)) {
+		LOG_ERR("%s device not ready", config->i2c_dev->name);
+		return -ENODEV;
+	}
+
+	if (IS_ENABLED(CONFIG_GPIO_NCT38XX_INTERRUPT)) {
+		nct38xx_init_interrupt(dev);
+	}
+
+	return 0;
+}
+
+#define GPIO_DEV_AND_COMMA(node_id) DEVICE_DT_GET(node_id),
+
+#define GPIO_NCT38XX_DEVICE_INSTANCE(inst)                                                         \
+	const struct device *sub_gpio_dev_##inst[] = { DT_FOREACH_CHILD_STATUS_OKAY(               \
+		DT_DRV_INST(inst), GPIO_DEV_AND_COMMA) };                                          \
+	static const struct gpio_nct38xx_config gpio_nct38xx_cfg_##inst = {                        \
+		.i2c_dev = DEVICE_DT_GET(DT_INST_BUS(inst)),                                       \
+		.i2c_addr = DT_INST_REG_ADDR(inst),                                                \
```

Comment:
Please use an i2c_dt_spec instead, and initialize it with I2C_DT_SPEC_INST_GET. Then you can use the `_dt` variants of other I2C routines to save boilerplate, like `i2c_reg_read_byte_dt` instead of `i2c_reg_read_byte`.

Reply:
Done.